engine/table/src/main/java/io/deephaven/engine/table/impl/hierarchical/TreeTableImpl.java

```diff
@@ -222,6 +222,28 @@ protected TreeTableImpl copy() {
                 parentIdentifierColumn, nodeFilterColumns, nodeOperations, availableColumnDefinitions);
     }
 
+    @Override
+    public TreeTable rebase(@NotNull final Table newSource) {
+        if (!newSource.getDefinition().equals(source.getDefinition())) {
+            if (newSource.getDefinition().equalsIgnoreOrder(source.getDefinition())) {
+                throw new IllegalArgumentException(
+                        "Cannot rebase a TreeTable with a new source definition, column order is not identical.");
+            }
+            final String differenceDescription = newSource.getDefinition()
+                    .getDifferenceDescription(source.getDefinition(), "new source", "existing source", ",");
+            throw new IllegalArgumentException(
+                    "Cannot rebase a TreeTable with a new source definition: " + differenceDescription);
+        }
+
+        final QueryTable newSourceQueryTable = (QueryTable) newSource;
+        final QueryTable tree = computeTree(newSourceQueryTable, parentIdentifierColumn);
+        final QueryTable sourceRowLookupTable = computeSourceRowLookupTable(newSourceQueryTable, identifierColumn);
+        final TreeSourceRowLookup sourceRowLookup = new TreeSourceRowLookup(newSource, sourceRowLookupTable);
```

Comment:
Shouldn't this be on the actual table?
```suggestion
        final TreeSourceRowLookup sourceRowLookup = new TreeSourceRowLookup(newSourceQueryTable, sourceRowLookupTable);
```